tests/unit-core/test-api-strict-equal.c

```diff
@@ -0,0 +1,63 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+
+#include "test-common.h"
+
+int
+main (void)
+{
+  TEST_INIT ();
+
+  jerry_init (JERRY_INIT_EMPTY);
+
+  /* Test objects */
+  jerry_value_t obj_val = jerry_eval ((const jerry_char_t *) "o={x:1};o", 9, JERRY_PARSE_NO_OPTS);
```

Comment:
There should be way more tests for this api. To make it easier we could do something like this:
```c
typedef struct {
  jerry_value_t lhs;
  jerry_value_t rhs;
  bool should_be_equal; 
} test_entry;

test_entry tests[] = {
 { jerry_create_number(3.1), jerry_create_number(10), true },
 ....
};

for (int idx = 0; idx < sizeof(tests) / sizeof(test_entry); idx++) {
  bool result = jerry_value_strict_equal (tests[idx].lhs, tests[idx].rhs);
  if (result !=  tests[idx].should_be_equal) {
    fprintf (stderr, "Test case failed, idx: %d\n", idx);
    TEST_ASSERT (0);
  }
  jerry_release_value (tests[idx].lhs);
  jerry_release_value (tests[idx].rhs);
}
```
or something like this.

A few missing test cases:
* string vs any other string (which are different)
* array vs any other thing
* typedarray vs any other thing
* is this true?  `jerry_value_strict_equal(a,b)` <=> `jerry_value_strict_equal(b,a)`
* error objects
* arraybuffers
* symbols